### PR TITLE
Adding podSecurityContextOverride

### LINKED ISSUE #110: Add supplementalGroups field to pod securityContext
Hello,

Would it be possible to add `supplementalGroups` to  the deployment template? We have a large PVC and every time  the pod is recreated, it takes 45-60 minutes to be reattached due to the usage of `fsGroup`. There are many issues about this problem, for example this one https://github.com/kubernetes/kubernetes/issues/67014

Up until kubernetes 1.17, specifying `supplementalGroups` instead of `fsGroup` is an  ok  workaround for this.

As of 1.18, an alpha feature has been added to specifically address this problem: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#configure-volume-permission-and-ownership-change-policy-for-pods. But it is still an alpha feature and is not likely to be enabled/permitted on multi-tenant production clusters. Therefore, allowing users of this helm chart to specify `supplementalGroups` would still be very helpful.

### LINKED ISSUE #203: Azure Managed Disk PersistentVolume taking up to 40 minutes to attach
**Describe the bug**
Azure Managed Disk PersistentVolume on AKS can take anywhere from 25 to 45 minutes to attach as the `jenkins_home`.

It started at 10 minutes and has slowly increased over the course of a year, spanning multiple versions of Kubernetes, and the Jenkins Helm Chart.

After having worked this out with the Azure Support team, it was determined that the issue is https://github.com/kubernetes/kubernetes/issues/67014.  Specifically, the solution is https://github.com/kubernetes/kubernetes/issues/67014#issuecomment-589915496.

`supplementalGroups` needs to be used instead of `fsGroup` in `securityContext`.

We have a PR coming shortly.

**Version of Helm and Kubernetes**:

Helm Version:

```console
$ helm version
version.BuildInfo{Version:"v3.4.2", GitCommit:"23dd3af5e19a02d4f4baa5b2f242645a1a3af629", GitTreeState:"clean", GoVersion:"go1.14.13"}
```

Kubernetes Version:

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.12", GitCommit:"5ec472285121eb6c451e515bc0a7201413872fa3", GitTreeState:"clean", BuildDate:"2020-09-16T13:39:51Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.13", GitCommit:"30d651da517185653e34e7ab99a792be6a3d9495", GitTreeState:"clean", BuildDate:"2020-10-20T16:22:33Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Which version of the chart**:
`main`, but has been happening since `v2`, maybe `v1`.

**What happened**:
Azure Managed Disk PersistentVolume on AKS can take anywhere from 25 to 45 minutes to attach as the `jenkins_home`.

**What you expected to happen**:
The PersistentVolume should be taking a fraction of the time to mount.

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by pedrohdz**
# What this PR does / why we need it

This PR introduces the `podSecurityContextOverride` value that completely overwrites `securityContext` in both the controller `StatefulSet` and backup `CronJob`.  It gives greater flexability in what can be put in the pod `securityContext` block, not limited to just `runAsUse`, `fsGroup`, and `securityContextCapabilities`.

In our particular case, we need to be able to unset `fsGroup` and set `supplementalGroups` instead.  Using an override as implemented, would allow use of other `securityContext` values without further complicating the existing Helm Chart.

This is related to the solution in [ kubernetes/kubernetes - Timeout expired waiting for volumes to attach #67014](https://github.com/kubernetes/kubernetes/issues/67014).

# Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #203
- fixes #110

# Special notes for your reviewer

# Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] CHANGELOG.md was updated

### CODE REVIEW

File: charts/jenkins/values.yaml [hunk 1]
```diff
@@ -78,7 +78,7 @@ controller:
   # jenkinsUrl: ""
   # If you set this prefix and use ingress controller then you might want to set the ingress path below
   # jenkinsUriPrefix: "/jenkins"
-  # Enable pod security context (must be `true` if runAsUser or fsGroup are set)
+  # Enable pod security context (must be `true` if podSecurityContextOverride, runAsUser or fsGroup are set)
   usePodSecurityContext: true
   # Set runAsUser to 1000 to let Jenkins run as non-root user 'jenkins' which exists in 'jenkins/jenkins' docker image.
   # When setting runAsUser to a different value than 0 also set fsGroup to the same value:
```
Comment (timja) [inline charts/jenkins/values.yaml:84]: probably should have a big warning that you should set supplemental groups instead

File: charts/jenkins/VALUES_SUMMARY.md [hunk 2]
```diff
@@ -99,9 +99,10 @@ The following tables list the configurable parameters of the Jenkins chart and t
 | `controller.resources`                | Resources allocation (Requests and Limits) | `{requests: {cpu: 50m, memory: 256Mi}, limits: {cpu: 2000m, memory: 4096Mi}}`|
 | `controller.initContainerEnv`         | Environment variables for Init Container                                 | Not set |
 | `controller.containerEnv`             | Environment variables for Jenkins Container                              | Not set |
-| `controller.usePodSecurityContext`    | Enable pod security context (must be `true` if `runAsUser` or `fsGroup` are set) | `true` |
+| `controller.usePodSecurityContext`    | Enable pod security context (must be `true` if `runAsUser`, `fsGroup`, or `podSecurityContextOverride` are set) | `true` |
```
Comment (timja) [inline charts/jenkins/VALUES_SUMMARY.md:102]: should we deprecate runAsUser and fsGroup?

File: charts/jenkins/values.yaml (hunk 1 above)
Comment (pedrohdz) [reply charts/jenkins/values.yaml:84]: Put the warning somewhere in the README?  FAQ file, or section in the README might might be a bad idea.

File: charts/jenkins/VALUES_SUMMARY.md (hunk 2 above)
Comment (pedrohdz) [reply charts/jenkins/VALUES_SUMMARY.md:102]: Add comments along the lines of deprecating them?  Completely pulling them out would be a major version bump.

**Comment by pedrohdz** [comment]
@timja ,
Updated per your comments.  Please review.

**Comment by pedrohdz** [comment]
@timja. @torstenwalter,
Any thoughts?  We could drop `podSecurityContextOverride` in favor of adding a `supplementalGroups` explicitly if that would help?  I am partial to the `podSecurityContextOverride` solution though since it gives more flexibility.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
> @timja. @torstenwalter,
> Any thoughts? We could drop `podSecurityContextOverride` in favor of adding a `supplementalGroups` explicitly if that would help? I am partial to the `podSecurityContextOverride` solution though since it gives more flexibility.

apologies meant to come back to this but forgot

**Pull request merged**